lms/static/scripts/frontend_apps/utils/api.js

```diff
@@ -90,6 +94,36 @@ async function apiCall({ path, authToken, data }) {
   return resultJson;
 }
 
-// Separate export from declaration to work around
-// https://github.com/robertknight/babel-plugin-mockable-imports/issues/9
-export { apiCall };
+/** @param {number} ms */
+function delay(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+/**
+ * Fetch a list of available ebooks to use in assignments.
+ *
+ * @param {string} authToken
+ * @param {number} [fetchDelay] - Dummy delay to simulate slow third-party
+ * @return {Promise<Book[]>}
+ */
+export async function fetchBooks(authToken, fetchDelay = 500) {
+  await delay(fetchDelay);
+  return bookList;
+}
+
+/**
+ * Fetch a list of chapters that can be used as the target location for an
+ * ebook assignment.
+ *
+ * @param {string} authToken
+ * @param {string} bookId
+ * @param {number} [fetchDelay] - Dummy delay to simulate slow third-party
+ * @return {Promise<Chapter[]>}
+ */
+export async function fetchChapters(authToken, bookId, fetchDelay = 500) {
+  await delay(fetchDelay);
+  if (!chapterData[bookId]) {
+    throw new ApiError(404, { message: 'Book not found' });
+  }
```

Comment:
Is this branch a stub for a "real" error handler or to exercise a test path? i.e. is this an expected possible path in the current implementation?